### PR TITLE
New Metaverse PR

### PR DESCRIPTION
**Pull request by kasenvr**
OK, so this is semi-tested...

Based on PR #691 .

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-6c4a024_61d12c66.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-6c4a024_61d12c66.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-6c4a024.tar_61d12c66.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-6c4a024.tar_61d12c66.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-6c4a024_61d12c66.exe

**Comment by daleglass** [comment]
I like the overall idea, but here's a thought I just had: what if instead of a constant we used a function call instead?

This would mean it could be changed without recompiling, the URL would be configurable, we could do things like load balancing between different servers, and the resulting URL could be sanitized (eg, S3 doesn't really parse paths, so "/foo/bar.fst" and "/foo//bar.fst" are really different files for it)

**Comment by kasenvr** [comment]
This PR will be updated once #691 is merged.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c1a8c2e.tar_7b7038fa.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c1a8c2e.tar_7b7038fa.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c1a8c2e_7b7038fa.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c1a8c2e_7b7038fa.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c1a8c2e_7b7038fa.exe

**Comment by Aitolda** [comment]
Will not even open for me.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9a76e75_5a321cda.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9a76e75.tar_5a321cda.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9a76e75.tar_5a321cda.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9a76e75_5a321cda.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9a76e75_5a321cda.exe

File: interface/resources/qml/hifi/avatarapp/AvatarsModel.qml [hunk 1]
```diff
@@ -49,7 +49,13 @@ ListModel {
     }
 
     function makeAvatarObject(avatar, avatarName) {
-        var avatarThumbnailUrl = makeThumbnailUrl(avatar.avatarUrl);
+        var avatarThumbnailUrl;
+        
+        if (!avatar.avatarIcon) {
+            makeThumbnailUrl(avatar.avatarUrl);
```
Comment (vegaslon) [inline interface/resources/qml/hifi/avatarapp/AvatarsModel.qml:55]: need to set this to avatarThumbnailUrl

Comment (kasenvr) [reply interface/resources/qml/hifi/avatarapp/AvatarsModel.qml:55]: Good catch! lol

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-d4f922e.tar_892f97d3.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-d4f922e_892f97d3.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-d4f922e.tar_892f97d3.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-d4f922e_892f97d3.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-d4f922e_892f97d3.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-b8d4034.tar_a50918b7.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-b8d4034_a50918b7.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-b8d4034.tar_a50918b7.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-b8d4034_a50918b7.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-b8d4034_a50918b7.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-a63c1f4.tar_2c1a51e7.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-a63c1f4.tar_2c1a51e7.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-a63c1f4_2c1a51e7.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-a63c1f4_2c1a51e7.dmg

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c4e4b61.tar_9e8dc88f.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c4e4b61.tar_9e8dc88f.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c4e4b61_9e8dc88f.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c4e4b61_9e8dc88f.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-c4e4b61_9e8dc88f.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9179db0.tar_46875a6c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9179db0.tar_46875a6c.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9179db0_46875a6c.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9179db0_46875a6c.exe

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR668/Vircadia-Alpha-PR668-9179db0_46875a6c.dmg

**Comment by kasenvr** [comment]
Merging, mostly CR'd, tested a decent bit, we'll catch the rest in testing. :)

**Pull request merged**